source/common/protobuf/utility.cc

```diff
@@ -12,10 +13,24 @@ MissingFieldException::MissingFieldException(const std::string& field_name,
           fmt::format("Field '{}' is missing in: {}", field_name, message.DebugString())) {}
 
 void MessageUtil::loadFromFile(const std::string& path, Protobuf::Message& message) {
-  const std::string json = Filesystem::fileReadToEnd(path);
-  const auto status = Protobuf::util::JsonStringToMessage(json, &message);
+  const std::string file_contents = Filesystem::fileReadToEnd(path);
+  // If the filename ends with .pb, do a best-effort attempt to parse it as a proto.
+  if (path.size() > 3 && path.substr(path.size() - 3) == ".pb") {
```

Comment:
BTW, you can use `StringUtil::endsWith` here for checking suffix; just noticed this when doing a similar path match check.